services/scan/src/simple_interpreter.ts

```diff
@@ -0,0 +1,211 @@
+      const [frontQrcodeOutput, backQrcodeOutput] =
+        qrcodeWorker.normalizeSheetOutput(
+          electionDefinition,
+          await mapSheet(
+            [frontPath, backPath],
+            qrcodeWorker.detectQrcodeInFilePath
+          )
+        );
+      return ok(
+        await mapSheet(
+          [
+            [frontPath, frontQrcodeOutput],
+            [backPath, backQrcodeOutput],
+          ] as const,
+          async ([
+            ballotImagePath,
+            detectQrcodeResult,
+          ]): Promise<PageInterpretationWithFiles> => {
+            const ballotImageFile = await readFile(ballotImagePath);
+            const result = await vxInterpreter.interpretFile({
+              ballotImagePath,
+              ballotImageFile,
+              detectQrcodeResult,
+            });
+
+            const images = await saveSheetImages(
+              sheetId,
+              ballotImagesPath,
+              ballotImagePath,
+              result.normalizedImage
+            );
+
+            return {
+              interpretation: result.interpretation,
+              originalFilename: images.original,
+              normalizedFilename: images.normalized,
+            };
+          }
+        )
+      );
+    },
+  };
+}
+
+/**
+ * Loads ballot layouts from the {@link Store} suitable for passing to {@link
+ * createInterpreter}. The results may be cached and used again as long as the
+ * underlying HMPB templates are not modified.
+ */
+export async function loadLayouts(
+  store: Store
+): Promise<BallotPageLayoutWithImage[] | undefined> {
+  const electionDefinition = store.getElectionDefinition();
+
+  if (!electionDefinition) {
+    return;
+  }
+
+  const templates = store.getHmpbTemplates();
+  const loadedLayouts: BallotPageLayoutWithImage[] = [];
+
+  for (const [pdf, layouts] of templates) {
+    for await (const { page, pageNumber } of pdfToImages(pdf, { scale: 2 })) {
+      const ballotPageLayout = layouts[pageNumber - 1];
+      loadedLayouts.push(
+        await interpretTemplate({
+          electionDefinition,
+          imageData: page,
+          metadata: ballotPageLayout.metadata,
+        })
+      );
+    }
+  }
+
+  return loadedLayouts;
+}
+
+/**
+ * Creates an interpreter with the given options. Use {@link loadLayouts} to
+ * load ballot layouts from the {@link Store}.
+ */
+export function createInterpreter(
+  options: CreateInterpreterOptions
+): SimpleInterpreter {
+  if (options.electionDefinition.election.gridLayouts) {
+    return createNhInterpreter(options);
+  }
+
+  return createVxInterpreter(options);
+}
+
+/**
+ * Stores an interpreted sheet of ballot paper in the {@link Store}, returning
+ * the effective sheet ID.
+ */
+export function storeInterpretedSheet(
+  store: Store,
+  sheetId: Id,
+  sheet: SheetOf<PageInterpretationWithFiles>
+): Id {
+  // TODO instead of one batch per ballot, use one batch per scanning session
+  // (e.g. from polls open to polls close)
```

Comment:
before you do that, keep in mind that we have code that removes an incomplete batch on startup of the scanning software.

Reply:
Hmm, yeah it will probably take some care to make this safe. It's not high on my todo list - prob only really important when we get to RLAs